src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +204,174 @@ public int draw() {
         }
         return normalizedHistogram;
     }
-}
+
+    /**
+     * Determines if a variant would be called under the particular conditions of a given total depth, alt depth,
+     * average base qualities, allele fraction of variant and log odds threshold necessary to exceed to call variant.
+     * @param totalDepth Depth at the site to be called, both alt and ref.
+     * @param altDepth Number of alt bases at this site.
+     * @param averageQuality Average Phred-scaled quality of bases
+     * @param alleleFraction Allele fraction we are attempting to detect
+     * @param logOddsThreshold Log odds threshold necessary to exceed for variant to be called
+     * @return
+     */
+    public static boolean isCalled(int totalDepth, int altDepth, double averageQuality, double alleleFraction, double logOddsThreshold) {
+        double threshold;
+        double sumOfQualities = altDepth * averageQuality;
+        threshold = 10.0 * (altDepth * Math.log10(1.0 / alleleFraction) + (totalDepth - altDepth) * Math.log10(1.0 / (1.0 - alleleFraction)) + logOddsThreshold);
+
+        return sumOfQualities > threshold;
+    }
+
+    /**
+     * Draw from a binomial distribution.
+     * @param trials Number of trials to perform
+     * @param p Probability of individual success
+     * @param uniformRNG Random number generator to use for making draw
+     * @return Number of total successes
+     */
+    public static int binomialDraw(final int trials, final double p, final Random uniformRNG) {
+        if (p > 1.0 || p < 0) {
+            throw new PicardException("Probabilities should be between 0 and 1, found value " + p + ".");
+        }
+
+        int successes = 0;
+        for (int i = 0; i < trials; i++) {
+            if (uniformRNG.nextDouble() < p) {
+                successes++;
+            }
+        }
+        return successes;
+    }
+
+    /**
+     * Calculates the theoretical sensitivity with a given Phred-scaled quality score distribution at a constant
+     * depth.
+     * @param depth Depth to compute sensitivity at
+     * @param qualityDistribution Phred-scaled quality score distribution
+     * @param logOddsThreshold Log odd threshold necessary to exceed for variant to be called
+     * @param sampleSize sampleSize is the total number of simulations to run
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @param randomSeed random number seed to use for random number generator
+     * @return
+     */
+    public static double sensitivityAtConstantDepth(final int depth, final double[] qualityDistribution, final double logOddsThreshold, final int sampleSize, final double alleleFraction, final int randomSeed) {
+        final RouletteWheel qualityRW = new RouletteWheel(trimDistribution(qualityDistribution));
+        final Random uniformRNG = new Random(randomSeed);
+
+        int altDepth = 0;
+        int calledVariants = 0;
+        for (int k = 0; k < sampleSize; k++) {
```

Comment:
using some properties of the qDist, (mean and stdev) you should be able to figure out at what point (altDepth+alleleFraction) you can ignore the details of the distribution and use the meanQuality directly to approximate sumOfQualities

Reply:
That's true, but this would be an additional approximation to speed things up when the runtime isn't very long now.  I think if we find speed to be an issue I'd be happy to revisit this.